src/main/java/com/yelp/nrtsearch/server/luceneserver/highlights/HighlightFetchTask.java

```diff
@@ -95,4 +112,30 @@ public void processHit(SearchContext searchContext, LeafReaderContext hitLeaf, B
   public double getTimeTakenMs() {
     return timeTakenMs.doubleValue();
   }
+
+  /**
+   * Verify each highlighted field is highlight-able.
+   *
+   * @param highlightV2 the v2 version of highlight settings
+   * @throws IllegalArgumentException if any field failed pass the verification.
+   */
+  private void verifyHighlights(HighlightV2 highlightV2) {
+    for (String fieldName : highlightV2.getFieldsList()) {
+      FieldDef field = indexState.getField(fieldName);
+      if (!(field instanceof TextBaseFieldDef)) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Field %s is not a text field and does not support highlights", fieldName));
+      }
+      if (!((TextBaseFieldDef) field).isSearchable()) {
+        throw new IllegalArgumentException(
+            String.format("Field %s is not searchable and cannot support highlights", fieldName));
+      }
+      if (!((TextBaseFieldDef) field).isStored()) {
+        throw new IllegalArgumentException(
+            String.format("Field %s is not stored and cannot support highlights", fieldName));
+      }
```

Comment:
I think this check is specific to fast vector highlighter (Unified does not necessarily need it) and can be moved to its class